tfx/examples/cifar10/cifar10_pipeline_native_keras.py

```diff
@@ -0,0 +1,205 @@
+from tfx.types.standard_artifacts import Model
+from tfx.types.standard_artifacts import ModelBlessing
+from tfx.utils.dsl_utils import external_input
+
+_pipeline_name = 'cifar10_native_keras'
+
+# This example assumes that CIFAR10 train set data is stored in
+# ~/cifar10/data/train, test set data is stored in ~/cifar10/data/test, and
+# the utility function is in ~/cifar10. Feel free to customize as needed.
+_cifar10_root = os.path.join(os.environ['HOME'], 'cifar10')
+_data_root = os.path.join(_cifar10_root, 'data')
+# Python module files to inject customized logic into the TFX components. The
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file = os.path.join(_cifar10_root, 'cifar10_utils_native_keras.py')
+
+# Path which can be listened to by the model server. Pusher will output the
+# trained model here.
+_serving_model_dir_lite = os.path.join(_cifar10_root, 'serving_model_lite',
+                                       _pipeline_name)
+
+# Directory and data locations.  This example assumes all of the images,
+# example code, and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
+_pipeline_root = os.path.join(_tfx_root, 'pipelines', _pipeline_name)
+# Sqlite ML-metadata db path.
+_metadata_path = os.path.join(_tfx_root, 'metadata', _pipeline_name,
+                              'metadata.db')
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text,
+                     serving_model_dir_lite: Text,
+                     metadata_path: Text,
+                     direct_num_workers: int) -> pipeline.Pipeline:
+  """Implements the CIFAR10 image classification pipeline using TFX."""
+  input_config = example_gen_pb2.Input(splits=[
+      example_gen_pb2.Input.Split(name='train', pattern='train/*'),
+      example_gen_pb2.Input.Split(name='eval', pattern='test/*')])
+
+  examples = external_input(data_root)
+
+  # Brings data into the pipeline.
+  example_gen = ImportExampleGen(input=examples, input_config=input_config)
+
+  # Computes statistics over data for visualization and example validation.
+  statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'])
+
+  # Generates schema based on statistics files.
+  schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True)
+
+  # Performs anomaly detection based on statistics and data schema.
+  example_validator = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=schema_gen.outputs['schema'])
+
+  # Performs .ations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=schema_gen.outputs['schema'],
+      module_file=module_file)
+
+  # We take the raw examples as the input to the trainer,
```

Comment:
This comment is now out-of-date right since we moved the data augmentation within the trainer?